packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Models/RoundTripModel.cs

```diff
@@ -4,12 +4,129 @@
+        /// <param name="requiredUnknown"> required unknown. </param>
+        /// <param name="requiredRecordUnknown"> required record of unknown. </param>
+        /// <param name="modelWithRequiredNullable"> this is a model with required nullable properties. </param>
+        /// <param name="requiredBytes"> Required bytes. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="requiredString"/>, <paramref name="requiredCollection"/>, <paramref name="requiredDictionary"/>, <paramref name="requiredModel"/>, <paramref name="requiredUnknown"/>, <paramref name="requiredRecordUnknown"/>, <paramref name="modelWithRequiredNullable"/> or <paramref name="requiredBytes"/> is null. </exception>
+        public RoundTripModel(string requiredString, int requiredInt, IEnumerable<string> requiredCollection, IDictionary<string, string> requiredDictionary, Thing requiredModel, System.BinaryData requiredUnknown, IDictionary<string, System.BinaryData> requiredRecordUnknown, ModelWithRequiredNullableProperties modelWithRequiredNullable, System.BinaryData requiredBytes)
+        {
+            if (requiredString == null)
+            {
+                throw new ArgumentNullException(nameof(requiredString));
+            }
+            if (requiredCollection == null)
+            {
+                throw new ArgumentNullException(nameof(requiredCollection));
+            }
+            if (requiredDictionary == null)
+            {
+                throw new ArgumentNullException(nameof(requiredDictionary));
+            }
+            if (requiredModel == null)
+            {
+                throw new ArgumentNullException(nameof(requiredModel));
+            }
+            if (requiredUnknown == null)
+            {
+                throw new ArgumentNullException(nameof(requiredUnknown));
+            }
+            if (requiredRecordUnknown == null)
+            {
+                throw new ArgumentNullException(nameof(requiredRecordUnknown));
+            }
+            if (modelWithRequiredNullable == null)
+            {
+                throw new ArgumentNullException(nameof(modelWithRequiredNullable));
+            }
+            if (requiredBytes == null)
+            {
+                throw new ArgumentNullException(nameof(requiredBytes));
+            }
+
+            RequiredString = requiredString;
+            RequiredInt = requiredInt;
+            RequiredCollection = requiredCollection.ToList();
+            RequiredDictionary = requiredDictionary;
+            RequiredModel = requiredModel;
+            IntExtensibleEnumCollection = new List<int>();
+            FloatExtensibleEnumCollection = new List<int>();
+            FloatFixedEnumCollection = new List<float>();
+            IntFixedEnumCollection = new List<int>();
+            RequiredUnknown = requiredUnknown;
+            RequiredRecordUnknown = requiredRecordUnknown;
+            OptionalRecordUnknown = new Dictionary<string, System.BinaryData>();
```

Comment:
Currently, the `ChangeTrackingList` and `ChangeTrackingDictionary` types have not been implemented. Therefore, I'm assigning the parameters to `List` and `Dictionary`. Once https://github.com/microsoft/typespec/issues/3324 is implemented, they will be changed to `ChangeTrackingList` and `ChangeTrackingDictionary`.